website/docs/spec/index.md

```diff
@@ -155,13 +157,17 @@ A Channel record defines an encoded stream of messages on a topic.
-| 2     | id               | uint16                | A unique identifier for this channel within the file.                                                       |
-| 2     | schema_id        | uint16                | The schema for messages on this channel. A schema_id of 0 indicates there is no schema for this channel.    |
-| 4 + N | topic            | String                | The channel topic.                                                                                          |
-| 4 + N | message_encoding | String                | Encoding for messages on this channel. The [well-known message encodings][message_encodings] are preferred. |
-| 4 + N | metadata         | `Map<string, string>` | Metadata about this channel                                                                                 |
+The `uuid` field may be used to associate channels between files. For example, an application that merges MCAP files
+may merge channels with the same non-nil `uuid` into the same output channel.
+
+| Bytes | Name             | Type                  | Description                                                                                                                                                                    |
+| ----- | ---------------- | --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| 2     | id               | uint16                | A unique identifier for this channel within the file.                                                                                                                          |
+| 2     | schema_id        | uint16                | The schema for messages on this channel. A schema_id of 0 indicates there is no schema for this channel.                                                                       |
+| 4 + N | topic            | String                | The channel topic.                                                                                                                                                             |
+| 4 + N | message_encoding | String                | Encoding for messages on this channel. The [well-known message encodings][message_encodings] are preferred.                                                                    |
+| 4 + N | metadata         | `Map<string, string>` | Metadata about this channel                                                                                                                                                    |
+| 16    | uuid             | uuid                  | A globally unique identifier for this channel. A [nil UUID](https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.7) indicates no globally unique identifier is available. |
```

Comment:
seems like this could be solved by using channel metadata. is there something that makes this special enough to be an explicitly named field?

Reply:
Being a first class citizen means that tooling can rely on this field. Metadata is better read as "user-data" that is unspecified.

Reply:
Do we have information about such a use case? What if I use something other than UUID to identify channels? 

I think we should have hard coded fields where there is a demonstrated need for cross-tool interoperability. If this is only to support needs of individual companies then using a named metadata field would be sufficient.

Reply:
I am not convinced that UUID is the appropriate term or requirement. I would describe it is an identifier that is stable across multiple separate mcap files (possibly via rolling recording).

I think this PR needs to be split up into separate proposals that have clear problem statements. Right now its not in a state that makes a case on what problems it is solving.